### PR TITLE
Tomas to master - context manager for db connection

### PR DESCRIPTION
**Pull request by Omastto1**
Added context manager 
Added inheritance between abstract db and mongo / postgres
Reformated code

### CODE REVIEW

**Comment by Omastto1** [comment]
@DovaX review please

**Review by DovaX — COMMENTED** [review]
This is not here by design - should be only here if you use context manager "with" clause @Omastto1

File: dbhydra/dbhydra_core.py [hunk 1]
```diff
@@ -1042,124 +1035,136 @@ def __init__(self,name="new_db",config_file="config.ini"):
...
+
+def dict_to_df(dictionary, column1, column2):
+    df = pd.DataFrame(list(dictionary.items()), columns=[column1, column2])
+    return (df)
+
+
+from contextlib import contextmanager
+
+
+@contextmanager
```
Comment (DovaX) [inline dbhydra/dbhydra_core.py:1166]: Wouldn't this be better as a natural part of every "DB" object? so put for example in AbstractDb?

Comment (Omastto1) [reply dbhydra/dbhydra_core.py]: good point, dont know why i did not do that in the first place
moved to abstract db

**Comment by Omastto1** [comment]
> This is not here by design - should be only here if you use context manager "with" clause @Omastto1

image

replaced with connect_to_db context manager

**Pull request merged**